docs: Reshape Endpoints & APIs concepts, archive deprecated streaming (#673)
serverpod/serverpod_docs: docs/06-concepts/01-working-with-endpoints/04-middleware.md

---
sidebar_label: Endpoint middleware

> description: Add middleware to the API server to intercept endpoint requests and responses for concerns such as logging, caching, and rate limiting.

Reviewer: Non-blocking: lead the `description` with the topic noun.

```suggestion
description: Middleware on the API server intercepts endpoint requests and responses for concerns such as logging, caching, and rate limiting.
```
